Changelog.md

```diff
@@ -17,6 +17,7 @@ Compiler Features:
 
 
 Bugfixes:
+* Commandline Interface: Fix possible inconsistency in subassembly IDs between target architectures in `--asm-json` output.
```

Comment:
This is not just the interface:

```suggestion
* Assembler: Fix different IDs being assigned to subassemblies nested more than one level away, resulting in inconsistent `--asm-json` output between target architectures.
```

Reply:
Also, this needs to be moved to 0.8.31 now.

Reply:
> This is not just the interface:

Well yeah, although assuming the ID space isn't exhausted, it does behave consistently inside of the assembler. I don't mind either way so let's go with your phrasing. Although I would minimally amend it:

`* Assembler: Fix not using a fixed-width type for IDs being assigned to subassemblies nested more than one level away, resulting in inconsistent `--asm-json` output between target architectures.`